// --------------------------------------------------------------------------------
// SPDX-FileCopyrightText: 2024-2025 Martin Jan KÃ¶hler and Harald Pretl
// Johannes Kepler University, Institute for Integrated Circuits.
//
// This file is part of KPEX 
// (see https://github.com/martinjankoehler/klayout-pex).
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program. If not, see <http://www.gnu.org/licenses/>.
// SPDX-License-Identifier: GPL-3.0-or-later
// --------------------------------------------------------------------------------
syntax = "proto3";

package kpex.layout;

import "kpex/geometry/shapes.proto";

message Location {
    enum Kind {
        LOCATION_KIND_UNSPECIFIED = 0;
        LOCATION_KIND_POINT = 1;
//        LOCATION_KIND_POLYGON = 2;
        LOCATION_KIND_BOX = 3;
//        LOCATION_KIND_REGION = 4;
    }

    Kind kind = 10;

    oneof location {
        kpex.geometry.Point point = 20;
//        kpex.geometry.Polygon polygon = 30;
        kpex.geometry.Box   box = 40;
//        kpex.geometry.Region region = 50;
    }
}
